File: src/main/java/net/ndrei/teslacorelib/inventory/BoundingRectangle.java
package net.ndrei.teslacorelib.inventory;

/**
 * Created by CF on 2016-12-23.
 */
public final class BoundingRectangle {
    private int x, y, width, height;

    public BoundingRectangle(int x, int y, int width, int height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }

    public int getLeft() { return this.x; }
    public int getTop() { return this.y; }
    public int getWidth() { return this.width; }
    public int getHeight() { return this.height; }
    public int getRight() { return this.getLeft() + this.getWidth(); }
    public int getBottom() { return this.getTop() + this.getHeight(); }
}

File: src/main/kotlin/net/ndrei/teslacorelib/inventory/BoundingRectangle.kt
package net.ndrei.teslacorelib.inventory

import net.ndrei.teslacorelib.gui.FluidTankPiece

/**
 * Created by CF on 2017-06-28.
 */
class BoundingRectangle(val left: Int, val top: Int, val width: Int, val height: Int) {
    val right: Int
        get() = this.left + this.width
    val bottom: Int
        get() = this.top + this.height

    val isEmpty
        get() = (this === EMPTY)

    companion object {
        val EMPTY = BoundingRectangle(0, 0, 0, 0)

        fun slots(left: Int, top: Int, columns: Int, rows: Int) =
            BoundingRectangle(left, top, columns * 18, rows * 18)

        fun fluid(left: Int, top: Int) =
            BoundingRectangle(left, top, FluidTankPiece.WIDTH, FluidTankPiece.HEIGHT)
    }
}
